ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Abstract.hs

```diff
@@ -137,29 +136,33 @@ class ( Show (ChainState    p)
 
   -- | Do we prefer the candidate chain over ours?
   --
-  -- Returns 'True' when we prefer the candidate over our chain.
+  -- Should return 'True' when we prefer the candidate over our chain.
   --
-  -- PRECONDITION: the candidate chain does not extend into the future.
+  -- We pass only the tips of the chains; for all consensus protocols we are
+  -- interested in, this provides sufficient context. (Ouroboros Genesis is
+  -- the only exception, but we will handle the genesis rule elsewhere.)
   --
-  -- Note: we make no assumptions about the anchor points of the fragments.
+  -- PRECONDITIONS:
   --
-  -- Note: we do not assume that the candidate fragments fork less than @k@
-  -- blocks back.
+  -- * The candidate chain does not extend into the future.
+  -- * The candidate must intersect with our chain within @k@ blocks from
+  --   our tip.
+  --
+  -- NOTE: An assumption that is quite deeply ingrained in the design of the
+  -- consensus layer is that if a chain can be extended, it always should (e.g.,
+  -- see the chain database spec in @ChainDB.md@). This means that any chain
+  -- is always preferred over the empty chain, and 'preferCandidate' does not
+  -- need (indeed, cannot) be called if our current chain is empty.
   preferCandidate :: CanSelect p hdr
                   => NodeConfig p
-                  -> AnchoredFragment hdr      -- ^ Our chain
-                  -> AnchoredFragment hdr      -- ^ Candidate
+                  -> hdr      -- ^ Our chain
+                  -> hdr      -- ^ Candidate
```

Comment:
Our chain -> Header at our chain's tip? Or something like that

Reply:
I don't follow?

Reply:
It's about the comment

Reply:
Ah, gotcha. Done.